src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -387,8 +387,16 @@ public BulkLoadRegisterSDFResponseDTO registerSdf(BulkLoadRegisterSDFRequestDTO
 				Parent parent;
 				SaltForm saltForm;
 				Lot lot;
+				boolean saltInStruct = false; 
+				boolean saltInSDF = false;
 				// attempt to strip salts
 				try {
+					// Check mol mappings for salt present
+					StrippedSaltDTO strippedSaltDTO = chemStructureService.stripSalts(inputMol);
+					if (!strippedSaltDTO.getSaltCounts().isEmpty() && strippedSaltDTO.getUnidentifiedFragments().size() == 1)
```

Comment:
Checking getSaltCounts is correct, but I think there is a better place to put this code. Let's walk through the `processForSaltStripping` function. I think all of our checks belong there.
https://github.com/mcneilco/acas-roo-server/blob/3e9036584776f6b61221959ee8703d8d0a024fce/src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java#L2358

Reply:
Alright sounds good!